lib/3scale/api/client.rb

```diff
@@ -36,6 +36,61 @@ def list_applications(service_id: nil)
 
       # @api public
       # @return [Hash]
+      # @param [Fixnum] id Application ID
+      def show_application(id)
+        find_application(id: id)
+      end
+
+      # @api public
+      # @return [Hash]
+      # @param [Fixnum] id Application ID
+      # @param [String] user_key Application User Key
+      # @param [String] application_id Application App ID
+      def find_application(id: nil, user_key: nil, application_id: nil)
+        params = { application_id: id, user_key: user_key, app_id: application_id }.reject { |_,value| value.nil? }
+        response = http_client.get('/admin/api/applications/find', params: params)
+        extract(entity: 'application', from: response)
+      end
+
+      # @api public
+      # @return [Hash] an Application
+      # @param [Fixnum] plan_id Application Plan ID
+      # @param [String] name Application Name
+      # @param [String] description Application Description
+      # @param [Hash] attributes Application Attributes
+      # @option attributes [String] :user_key Application User Key
+      # @option attributes [String] :application_id Application App ID
+      # @option attributes [String] :application_key Application App Key(s)
+      # @option attributes [String] :username User Name. Defaults to User Email.
+      def create_application(account_id, attributes = {}, plan_id: , name: , description: , **rest)
+        body = { plan_id: plan_id,
+                 name: name,
+                 description: description }.merge(attributes).merge(rest)
+        response = http_client.post("/admin/api/accounts/#{account_id}/applications", body: body)
+        extract(entity: 'application', from: response)
+      end
+
+      # @api public
+      # @return [Hash] an Account
+      # @param [String] name Account Name
+      # @param [String] email User Email
+      # @param [String] password User Password
+      # @param [Hash] attributes User and Account Attributes
+      # @option attributes [String] :account_plan_id Account Plan ID
+      # @option attributes [String] :service_plan_id Service Plan ID
+      # @option attributes [String] :application_plan_id Application Plan ID
+      # @option attributes [String] :username User Name. Defaults to User Email.
+      def signup(attributes = {}, name: , email: , password: , **rest)
+        body = { org_name: name,
+                 username: email,
```

Comment:
Correct. From my tests only name & username are required. Password is also not required.